crates/js-sys/src/lib.rs

```diff
@@ -425,6 +425,76 @@ extern "C" {
     pub fn unshift(this: &Array, value: &JsValue) -> u32;
 }
 
+/// Iterator returned by `Array::iter`
+#[derive(Debug, Clone)]
+pub struct ArrayIter<'a> {
+    start: u32,
+    end: u32,
```

Comment:
Instead of start/end could this perhaps use `Range<u32>` instead? That could also easily map methods to all of the iterator methods on `Range` as well

Reply:
Done.

Reply:
I wouldn't have thought of using `Range` like that, guess you really do learn something new every day.